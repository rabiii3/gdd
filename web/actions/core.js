export const START_REQUEST = 'core/START_REQUEST';
export const END_REQUEST = 'core/END_REQUEST';
export const REQUEST_ERROR = 'core/REQUEST_ERROR';
export const ACK_ERROR = 'core/ACK_ERROR';
export const COMM_ERROR = 'commError';
export const AUTH_ERROR = 'authError';

export const startRequest = () => ({ type: START_REQUEST });
export const endRequest = () => ({ type: END_REQUEST });

export const requestError = ({ method, type, statusCode }) => ({
  type: REQUEST_ERROR,
  error: { method, type, statusCode },
});

export const ackError = () => ({ type: ACK_ERROR });
