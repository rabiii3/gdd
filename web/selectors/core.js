import { pathEq, prop } from 'ramda';
import { createSelector } from 'reselect';

export const root = prop('core');
export const hasSignInError = createSelector(root, pathEq(['error', 'method'], 'auth:signIn'));
export const hasRegisterError = createSelector(root, pathEq(['error', 'method'], 'auth:register'));
export const hasRegisterInError = createSelector(root, pathEq(['error', 'method'], 'auth:registerIn'));
export const getError = createSelector(root, prop('error'));
