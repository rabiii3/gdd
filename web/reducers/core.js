import { pathOr } from 'ramda';
import { AUTH_ERROR, COMM_ERROR, START_REQUEST, END_REQUEST, REQUEST_ERROR, ACK_ERROR } from '../actions/core';

const getErrorType = ({ statusCode }) => {
  if (statusCode === 401 || statusCode === 403) return AUTH_ERROR;
  return COMM_ERROR;
};

const initialState = { pendingRequests: 0 };
const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case START_REQUEST:
      return {
        ...state,
        pendingRequests: state.pendingRequests + 1,
      };
    case REQUEST_ERROR:
      return {
        ...state,
        error: { ...action.error, id: pathOr(0, ['error', 'id'], state) + 1, type: getErrorType(action.error) },
        pendingRequests: state.pendingRequests - 1,
      };
    case END_REQUEST:
      return {
        ...state,
        pendingRequests: state.pendingRequests - 1,
      };
    case ACK_ERROR:
      return {
        ...state,
        error: null,
      };
    default:
      return state;
  }
};

export default reducer;
