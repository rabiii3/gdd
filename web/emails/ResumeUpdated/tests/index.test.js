import React from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { Provider } from 'react-redux';
import configureStore from '../../../store/configureStore.test';
import { shallow } from 'enzyme';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import { enhance } from '..';

describe('web |Â pages | EmailConfirmation | index', () => {
  describe('render', () => {
    it('should emailConfirmation render', () => {
      const theme = createMuiTheme({
        typography: {
          useNextVariants: true,
        },
      });
      const Component = () => <div />;
      const EnhancedComponent = enhance(Component);
      const history = createBrowserHistory();
      const state = { language: { locale: 'en' } };
      const store = configureStore(state);

      const wrapper = shallow(
        <MuiThemeProvider theme={theme}>
          <Provider store={store}>
            <Router history={history}>
              <EnhancedComponent />
            </Router>
          </Provider>
        </MuiThemeProvider>,
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
