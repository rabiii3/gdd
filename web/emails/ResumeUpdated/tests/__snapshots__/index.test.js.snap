// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web |Â pages | EmailConfirmation | index render should emailConfirmation render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <WithStyles(Component) />
  </Router>
</Provider>
`;
